hibernate-core/src/main/java/org/hibernate/query/internal/QueryParameterBindingsImpl.java

```diff
@@ -528,6 +527,25 @@ public String expandListValuedParameters(String queryString, SharedSessionContra
 			final QueryParameter sourceParam = entry.getKey();
 			final Collection bindValues = entry.getValue().getBindValues();
 
+			int bindValueCount = bindValues.size();
+			int bindValueMaxCount = bindValueCount;
+
+			boolean inClauseParameterPaddingEnabled =
+					session.getFactory().getSessionFactoryOptions().inClauseParameterPaddingEnabled() &&
+					bindValueCount > 2;
+
+			if ( inClauseParameterPaddingEnabled ) {
```

Comment:
Not sure which is ultimately faster, but have you tried `final int bindValueMaxCount = (int) (1L << -Long.numberOfLeadingZeros(count - 1) );`?

Reply:
BTW, I got that from Guava Math classes

Reply:
Also, no matter which way we choose, I'd much prefer this be moved into a method.  The intent is not very clear in either approach - but calling a method like `#ceilingPowerOfTwo` (Guava's name) is much more readable, imo

Reply:
I'll move it to some Utils class and check which one is faster as well.